docs/contributors/code/e2e/overusing-snapshots.md

```diff
@@ -0,0 +1,133 @@
+
+Here's the same test with the test title and comments. Now you know what these assertions are actually about.
+
+```js
+it( 'can be split at the end', async () => {
+	// ...
+
+	// Expect empty paragraph outside quote block.
+	expect( await getEditedPostContent() ).toMatchSnapshot();
+
+	// ...
+
+	// Expect the paragraph to be merged into the quote block.
+	expect( await getEditedPostContent() ).toMatchSnapshot();
+} );
+```
+
+The developer's intention is a bit more readable, but it still feels disconnected from the test. You might be tempted to try [inline snapshots](https://jestjs.io/docs/snapshot-testing#inline-snapshots), which does solve the issue of having to jump around files, but it's still not self-documented nor explicit. We can do better.
+
+## The solution
+
+Instead of writing the assertions in comments, we can try directly writing them out explicitly. With the help of `editor.getBlocks`, we can rewrite them into simpler and atomic assertions.
+
+```js
+// ...
+
+{
+	// Expect empty paragraph outside quote block.
+	const blocks = await editor.getBlocks();
+	expect( blocks ).toHaveLength( 2 );
+	expect( blocks[ 0 ] ).toMatchObject( {
+		name: 'core/quote',
+		innerBlocks: [
+			{
+				name: 'core/paragraph',
+				attributes: { content: '1' },
+			},
+		],
+	} );
+	expect( blocks[ 1 ] ).toMatchObject( {
+		name: 'core/paragraph',
+		attributes: { content: '' },
+	} );
+}
+
+// ...
+
+{
+	// Expect the paragraph to be merged into the quote block.
+	const blocks = await editor.getBlocks();
+	expect( blocks ).toHaveLength( 1 );
+	expect( blocks[ 0 ] ).toMatchObject( {
+		name: 'core/quote',
+		innerBlocks: [
+			{
+				name: 'core/paragraph',
+				attributes: { content: '1' },
+			},
+			{
+				name: 'core/paragraph',
+				attributes: { content: '2' },
+			},
+		],
+	} );
+}
+```
+
+The assertions are more readable and explicit. You can add additional assertions or split existing ones into multiple ones to highlight their importance. Whether to keep the comments is up to you, but it's usually fine to omit them when the code is already readable without them.
```

Comment:
```suggestion
These assertions are more readable and explicit. You can add additional assertions or split existing ones into multiple ones to highlight their importance. Whether to keep the comments is up to you, but it's usually fine to omit them when the code is already readable without them.
```

Should we make it clearer that we're talking about the assertions in the example above or the "atomic assertions"?

Reply:
In my mind, they are atomic assertions. It's a term I coined for assertions that test minimal things 😅. Open to ideas for better namings/descriptions though!